src/components/tasks/card/index.tsx

```diff
@@ -1,160 +1,209 @@
+  onContentChange = () => undefined,
+}) => {
+  const statusRedList = [BLOCKED];
+  const statusNotOverDueList = [COMPLETED, VERIFIED, AVAILABLE];
+  const cardDetails = content;
+  const [assigneeProfilePic, setAssigneeProfilePic] = useState(
+    `${process.env.NEXT_PUBLIC_GITHUB_IMAGE_URL}/${cardDetails.assignee}/img.png`,
+  );
+  const contributorImageOnError = () => setAssigneeProfilePic('/dummyProfile.png');
+
+  const localStartedOn = inputParser(cardDetails.startedOn);
+  const fromNowStartedOn = moment(localStartedOn).fromNow();
+
+  const localEndsOn = inputParser(cardDetails.endsOn);
+  const fromNowEndsOn = moment(localEndsOn).fromNow();
+  const statusFontColor = !statusRedList.includes(cardDetails.status) ? '#00a337' : '#f83535';
+  const iconHeight = '25px';
+  const iconWidth = '25px';
+
+  const cardClassNames = [classNames.card];
+
+  function isTaskOverdue() {
+    const timeLeft = localEndsOn.valueOf() - Date.now();
+    return !statusNotOverDueList.includes(cardDetails.status) && timeLeft <= 0;
+  }
+  
+  function stripHtml(html: string) {
+    const tmp = document.createElement('DIV');
+    tmp.innerHTML = html;
+    return tmp.textContent || tmp.innerText || '';
+  }
+  
+  function handleChange(event: any, changedProperty: keyof typeof cardDetails) {
+    if (event.key === 'Enter') {
+      const toChange: any = cardDetails;
+      toChange[changedProperty] = stripHtml(event.target.innerHTML);
+      
+      if (changedProperty === 'endsOn' || changedProperty === 'startedOn') {
+        const toTimeStamp = new Date(`${toChange[changedProperty]}`).getTime() / 1000;
+        toChange[changedProperty] = toTimeStamp;
+      }
+      
+      onContentChange(toChange.id, {
+        [changedProperty]: toChange[changedProperty],
+      });
+    }
+  }
+
+  function inputParser(input: string) {
+    const parsedDate = moment(new Date(parseInt(input, 10) * 1000))
+    return parsedDate
+  }
+
+  function getPercentageOfDays(startedOn: string, endsOn: string): number {
+    const startDate = inputParser(startedOn)
+    const endDate = inputParser(endsOn)
+
+    // It provides us with total days that are there for the the project and number of days left
+    const totalDays = endDate.diff(startDate, 'days')
+    const daysLeft = endDate.diff(new Date(), 'days')
+
+    // It provides the percentage of days left
+    const percentageofDays = daysLeft/totalDays * 100
+    return percentageofDays
+  }
+  
+  function handleProgressColor(percentCompleted: number, startedOn: string, endsOn: string): string {
+    const percentageofDays = getPercentageOfDays(startedOn, endsOn)
+
+    if(percentageofDays >= percentCompleted) {
+      return classNames.progressGreen
+    } else if(percentageofDays < 50 && percentCompleted > 75) {
+      return classNames.progressOrange
+    } else if(percentageofDays < 25 && percentCompleted > 35) {
+      return classNames.progressRed
+    } else {
+      return classNames.progressYellow
+    }

```

Comment:
Nit changes to make the code more 'linear' to read: 

```suggestion
    if(percentageofDays >= percentCompleted) {
      return classNames.progressGreen
    }
    
    if(percentageofDays < 50 && percentCompleted > 75) {
      return classNames.progressOrange
    }
    
    if(percentageofDays < 25 && percentCompleted > 35) {
      return classNames.progressRed
    }
    
    return classNames.progressYellow;
```